24.12 Antalya Grype Workflow

strtgbb opened the pull request:
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

Adding a workflow for Grype scanning

#### Exclude tests:
- [ ] <!---ci_exclude_fast--> Fast test
- [x] <!---ci_exclude_integration--> Integration Tests
- [x] <!---ci_exclude_stateless--> Stateless tests
- [ ] <!---ci_exclude_stateful--> Stateful tests
- [x] <!---ci_exclude_performance--> Performance tests
- [x] <!---ci_exclude_asan--> All with ASAN
- [x] <!---ci_exclude_tsan--> All with TSAN
- [x] <!---ci_exclude_msan--> All with MSAN
- [x] <!---ci_exclude_ubsan--> All with UBSAN
- [x] <!---ci_exclude_coverage--> All with Coverage
- [x] <!---ci_exclude_aarch64--> All with Aarch64
- [x] <!---ci_exclude_regression--> All Regression
- [x] <!---no_ci_cache--> Disable CI Cache

strtgbb committed:
workflow for grype scanning

MyroTk commented on .github/grype/parse_vulnerabilities_grype.py:
We probably don't need the copyright notice

MyroTk commented on .github/workflows/grype_scan.yml:
```diff
@@ -0,0 +1,62 @@
+        runs-on: [self-hosted, x86, type-cx42, image-x86-app-docker-ce]
+        steps:
+        - name: Checkout repository
+          uses: actions/checkout@v2
```
actions/checkout@v2, you can use v4

MyroTk commented on .github/workflows/grype_scan.yml:
```diff
@@ -0,0 +1,62 @@
+          run: |
+            python3 ./tests/ci/version_helper.py | tee /tmp/version_info
+            source /tmp/version_info
+            echo "::set-output name=docker_image::${{ github.event.inputs.docker_image }}:${{ github.event.pull_request.number }}-$CLICKHOUSE_VERSION_STRING"
```
Releases are done from a tag `push` event, so `event.pull_request.number` will not work. Should be ok if you default it to 0

strtgbb replied:
Maybe I should just set it to zero all the time?

strtgbb committed:
Cleanup and fix default pr number and upload artifacts

Enmk commented on .github/workflows/grype_scan.yml:
```diff
@@ -0,0 +1,71 @@
+    workflow_dispatch:
+        inputs:
+            docker_image:
+                description: 'Docker image with tag to scan'
```
why without a tag? What version is going to be scanned then?

strtgbb replied:
Tag is required when this workflow is called standalone. When this workflow is called by another workflow, the tag is determined from the repo with `tests/ci/version_helper.py`. That gives this workflow the same behaviour as `tests/ci/docker_server.py` when called without the `--version` argument, as is done by `tests/ci/ci_definitions.py`.

Enmk replied:
it would be nice to mention that in comments and/or input description

strtgbb replied:
I've added comments and made tag optional for both inputs. I don't know what will happen in a manual run with no tag specified.

strtgbb committed:
fix grype scan tags and add comments

try fixing tags again

strtgbb committed:
add skipping support to regression

strtgbb committed:
add missing apt update

fix path to grype upload script

Fix report command and workflow issues

strtgbb committed:
fix copilot's follies

fix version getter for grype scan

fix another copilot mistake

fix pr num in version

add tag for image to access awscli

smaller instance for grype too

strtgbb committed:
install awscli with pip

strtgbb committed:
add aws secrets

strtgbb committed:
Combine grype jobs with matrix, clean docker image name for log upload

strtgbb committed:
Add Grype results to ci_running

strtgbb committed:
fix version of actions/github-script

strtgbb committed:
fix

strtgbb committed:
fix

strtgbb committed:
fix

strtgbb committed:
fix

strtgbb committed:
fix

strtgbb committed:
add grype step summary and fix

strtgbb committed:
log grype's detected distro

strtgbb committed:
better summary

strtgbb commented:
Currently, the state of the Grype job in ci_running will always be `success`. Do we want to have it fail if high/severe issues are found?

strtgbb committed:
Add scan for alpine server

strtgbb committed:
Add high/critical stats to grype job outputs

strtgbb committed:
Merge branch 'antalya' into 24.12_grype_workflow

strtgbb committed:
Make parse_vulnerabilities_grype status more reliable

strtgbb committed:
add cve results to combined report

MyroTk committed:
Merge branch 'antalya' into 24.12_grype_workflow

strtgbb committed:
fix report errors

strtgbb committed:
fix quoting of backticks

strtgbb committed:
Fix and upgrade cve tables

strtgbb committed:
fix call to list_objects_v2

strtgbb committed:
fixes and tweaks

strtgbb committed:
add arrows to indicate that columns are sortable

strtgbb committed:
tweak summary

strtgbb committed:
fix pandas errors

strtgbb committed:
fix finishcheck

strtgbb committed:
Merge branch 'antalya' into 24.12_grype_workflow

strtgbb committed:
hotfix ci test report

strtgbb committed:
make sure github token is available

strtgbb committed:
Merge branch 'antalya' into fixes/independent_report

strtgbb committed:
tweaks

strtgbb committed:
update css

strtgbb committed:
sortable tables

strtgbb committed:
cleanup

strtgbb committed:
Merge branch 'antalya' into fixes/independent_report

strtgbb committed:
Merge branch 'antalya' into fixes/independent_report

strtgbb committed:
Merge branch 'fixes/independent_report' into 24.12_grype_workflow

strtgbb commented:
Merge #724 first
